src/stores/AccountController.ts

```diff
@@ -342,6 +342,35 @@ export const updateSubscription = async (status: 'active' | 'cancelled'): Promis
   });
 };
 
+export const updateCardDetails = async ({
+  cardName,
+  cardNumber,
+  cvc,
+  expMonth,
+  expYear,
+  currency,
+}: {
+  cardName: string;
+  cardNumber: string;
+  cvc: number;
+  expMonth: number;
+  expYear: number;
+  currency: string;
+}) => {
+  return await useAccount(async ({ customerId }) => {
+    return await useService(async ({ subscriptionService, sandbox = true }) => {
+      const response = await subscriptionService?.updateCardDetails({ cardName, cardNumber, cvc, expMonth, expYear, currency }, sandbox);
+      const activePayment = (await subscriptionService?.getActivePayment({ sandbox, customerId })) || null;
+      useAccountStore.setState({ loading: true });
+      useAccountStore.setState({
+        loading: false,
+        activePayment,
+      });
```

Comment:
Is this correct (2x setState)?

Reply:
935748a0862d1a60e85231924c272b4981f42060